CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
+  "uint8", // default
+  "float16",
 };
 
 // Feature detection:
-
 interface CanvasColorSpace {
   const CanvasColorSpaceEnum srgb = "srgb";
+  const CanvasColorSpaceEnum displayP3 = "display-p3";
   const CanvasColorSpaceEnum rec2020 = "rec-2020";
 };
 
-interface CanvasColorEncoding {
-  const CanvasColorEncodingEnum unorm8 = "unorm8";
-  const CanvasColorEncodingEnum unorm8Srgb = "unorm8-srgb";
-  const CanvasColorEncodingEnum float16 = "float16";
+interface CanvasColorType {
+  const CanvasColorTypeEnum uint8 = "uint8";
+  const CanvasColorTypeEnum float16 = "float16";
 };
+</pre>
 
-// Feature activation:
+Color spaces match their respective counterparts as defined in the [CSS colorspaces](https://www.w3.org/TR/css-color-4/#predefined). Support for color spaces should not be limited based on the capabilities of the user's display. There exist [Media Query APIs](https://www.w3.org/TR/mediaqueries-4/) for that purpose.
 
-partial dictionary CanvasRenderingContext2DSettings {
-  CanvasColorSpaceEnum colorSpace = "srgb";
-  CanvasColorEncodingEnum colorEncoding = "unorm8";
-};
+When displaying a canvas element, the browser must ensure that the colors displayed on the user's screen match the colors specified in the canvas element as closely as is possible on the platform. The color space conversions responsible for this happen in compositing, and should have no script-visible side-effects.
 
-partial dictionary WebGLContextAttributes {
-  CanvasColorSpaceEnum colorSpace = "srgb";
-  CanvasColorEncodingEnum colorEncoding = "unorm8";
-};
+Some colorType values will allow the user to specify floating-point colors outside of the usual [0, 1] interval. These values will be clamped to the range [0, 1] when being displayed.
+
+TODO: Add 'extended-srgb' and 'extended-srgb-linear' color spaces which will not be subject to this constraint.
+
+TODO: Add a mechanism for indicating that extended color spaces are to be used for HDR rendering (this should probably not be the default, because HDR rendering has substantial power costs).
+
+### HTMLCanvasElement API Changes
 
-dictionary ImageDataColorSettings {
-  CanvasColorSpaceEnum colorSpace = "srgb";
-  CanvasColorEncodingEnum colorEncoding = "unorm8";
+The behavior of the HTMLCanvasElement functions toDataURL and toBlob methods are to produce encodings that match the canvas element's color space and pixel depth as closely as possible, subject to the limitations of the encoding format. It may be appropriate to add additional parameters to the relevant APIs to allow for more user control in this area.
+
+### CanvasRenderingContext2D
+
+IDL Additions:
+<pre>
+partial dictionary CanvasRenderingContext2DSettings {
+  CanvasColorSpace colorSpace = "srgb";
+  CanvasColorType colorType = "uint8";
 };
 
 partial interface CanvasRenderingContext2D {
   CanvasRenderingContext2DSettings getContextAttributes();
+  ImageData getImageData(long sx, long sy, long sw, long sh, optional ImageDataSettings imageDataSettings);
```

Comment:
I do not see `ImageDataSettings` mentioned anywhere in this document besides this IDL definition.  

Please add a definition in this document along with a section on why it is needed and the behavior of `getImageData/createImageData` when passed the settings object.

Reply:
I will create a separate patch on ImageDataSettings (it was there in the past, maybe it got lost in a shuffle).